requests/models.py

```diff
@@ -363,8 +364,8 @@ def prepare_url(self, url, params):
 
         # Only want to apply IDNA to the hostname
         try:
-            host = host.encode('idna').decode('utf-8')
-        except UnicodeError:
+            host = idna.encode(host).decode('utf-8')
```

Comment:
Are you sure? I just ran `idna.encode('GooGo.co')` and there were no problems there. Can you provide a sample hostname to encounter the issue?